src/shared/utils/url.js

```diff
@@ -28,6 +29,13 @@ export function updateQuery(update) {
     if (_.isUndefined(value)) delete query[key];
     else query[key] = value;
   });
+  if (!!query.filter && !!query.filter.startDate
+      && moment(query.filter.startDate).isValid() === false) {
+    delete query.filter.startDate;
+  } else if (!!query.filter && !!query.filter.endDate
+      && moment(query.filter.endDate).isValid() === false) {
+    delete query.filter.endDate;
+  }
```

Comment:
No, you was not supposed to modify `updateQuery(..)`
what I was talking about was to make such checks in `src/shared/routes/Topcoder/ChallengeListing.jsx`
and if you see we need to remove these query params, then you do it by calling `updateQuery({ startDate: undefined, endDate: undefined })`. The `updateQuery(..)` method is just an auxiliary function for setting / updating / deleting query params from URL. That's it.